Apps/Gateway/src/ClientApp/src/stores/user.ts

```diff
@@ -0,0 +1,468 @@
+                .catch((resultError: ResultError) => {
+                    handleError(
+                        resultError,
+                        ErrorType.Update,
+                        ErrorSourceType.User
+                    );
+                    reject(resultError);
+                });
+        });
+    }
+
+    function closeUserAccount(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            userProfileService
+                .closeAccount(user.value.hdid)
+                .then((userProfile) => {
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    setUserError(resultError.resultMessage);
+                    reject(resultError);
+                });
+        });
+    }
+
+    function recoverUserAccount(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            userProfileService
+                .recoverAccount(user.value.hdid)
+                .then((userProfile) => {
+                    logger.debug(
+                        `recoverUserAccount: ${JSON.stringify(userProfile)}`
+                    );
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    setUserError(resultError.resultMessage);
+                    reject(resultError);
+                });
+        });
+    }
+
+    function retrieveEssentialData(): Promise<void> {
+        return new Promise((resolve) => {
+            status.value = LoadStatus.REQUESTED;
+            patientService
+                .getPatient(user.value.hdid)
+                .then((result: Patient) => {
+                    if (!result) {
+                        logger.debug("Patient retrieval failed");
+                        patientRetrievalFailed.value = true;
+                        resolve();
+                        return;
+                    }
+                    setPatient(result);
+                    userProfileService
+                        .getProfile(user.value.hdid)
+                        .then((userProfile) => {
+                            logger.debug(
+                                `User Profile: ${JSON.stringify(userProfile)}`
+                            );
+                            setProfileUserData(userProfile);
+                            resolve();
+                        })
+                        .catch((resultError: ResultError) => {
+                            logger.debug(
+                                `User Profile retrieval failed, status (${resultError.statusCode})`
+                            );
+                            handleError(
+                                resultError,
+                                ErrorType.Retrieve,
+                                ErrorSourceType.Profile
+                            );
```

Comment:
```suggestion
                            if (error.statusCode === 429) {
                                logger.debug(
                                    "User profile retrieval failed because of too many requests"
                                );
                                appStore.setAppError(AppErrorType.TooManyRequests);
                            } else {
                                logger.debug("User profile retrieval failed");
                                appStore.setAppError(AppErrorType.General);
                            }
```

This should be dispatching the app-wide errors.